Symfony/CS/Tokenizer/TokensAnalyzer.php

```diff
@@ -210,6 +210,73 @@ public function isArrayMultiLine($index)
         return false;
     }
 
+    public function getFunctionAttributes($index)
+    {
+        $tokens = $this->tokens;
+        $token = $tokens[$index];
+
+        if (!$token->isGivenKind(T_FUNCTION)) {
+            throw new \LogicException('No T_FUNCTION at given index');
+        }
+
+        $attributes = array(
+            'visibility' => null,
+            'static' => false,
+            'abstract' => false,
+            'final' => false,
+        );
+
+        for ($i = $index; $i >= 0; --$i) {
+            $token = $tokens->getPrevMeaningfulToken($i);
+            if (null === $token) {
+                break;
+            }
+
+            $i = $token;
+            $token = $tokens[$token];
+
+            if ($token->isGivenKind(array(T_WHITESPACE, T_COMMENT, T_DOC_COMMENT))) {
+                continue;
+            }
+
+            if ($token->isGivenKind(array(T_STATIC))) {
+                $attributes['static'] = true;
+                continue;
+            }
+
+            if ($token->isGivenKind(array(T_FINAL))) {
+                $attributes['final'] = true;
+                continue;
+            }
+
+            if ($token->isGivenKind(array(T_ABSTRACT))) {
+                $attributes['abstract'] = true;
+                continue;
+            }
+
+            // visibility
+
+            if ($token->isGivenKind(array(T_PRIVATE))) {
+                $attributes['visibility'] = T_PRIVATE;
+                continue;
+            }
+
+            if ($token->isGivenKind(array(T_PROTECTED))) {
+                $attributes['visibility'] = T_PROTECTED;
+                continue;
+            }
+
+            if ($token->isGivenKind(array(T_PUBLIC))) {
+                $attributes['visibility'] = T_PUBLIC;
+                continue;
+            }
+
+            break;
```

Comment:
Sure thing, please see https://github.com/FriendsOfPHP/PHP-CS-Fixer/commit/6bb6b9c4b3e775ca601d6f2b6e610e1906594d66 (let me know if this can be done better)